include/zephyr/drivers/flash.h

```diff
@@ -583,6 +583,84 @@ static inline const struct flash_parameters *z_impl_flash_get_parameters(const s
 __syscall int flash_ex_op(const struct device *dev, uint16_t code,
 			  const uintptr_t in, void *out);
 
+/**
+ * @brief Check if two flash regions on the same device overlap.
+ *
+ * @param a_start offset of first region.
+ * @param a_size size of first region.
+ * @param b_start offset of second region.
+ * @param b_size size of second region.
+ *
+ * @retval 0 if the regions do not overlap.
+ * @retval 1 if the regions overlap.
+ * @retval -EINVAL if any value provided is negative.
+ * @retval -EOVERFLOW if either region overflows the address space.
+ */
+int flash_regions_overlap(off_t a_start, off_t a_size, off_t b_start, off_t b_size);
+
+/**
+ * @brief Calculate the CRC of a region of flash memory.
+ *
+ * Calculate the CRC of a region of memory in flash using the IEEE CRC32
+ * algorithm.
+ *
+ * The caller may optionally supply a buffer of suitable size. If no buffer is
+ * provided, an internal buffer 16 bytes in size is used.
+ *
+ * @param dev Flash device.
+ * @param offset Offset of the region.
+ * @param size Size of the region.
+ * @param[out] buf Optional buffer to use or NULL.
+ * @param buf_size Optional buffer size or 0.
+ * @param[out] crc Pointer to the CRC value.
+ *
+ * @retval 0 on success.
+ * @retval -EINVAL if @a offset or @a size is negative, or if @a crc is NULL,
+ * or if @a buf_size is not a multiple of uint32_t size.
+ * @retval -EIO if an I/O error occurs while reading flash memory.
+ * @retval -ENODEV if @a dev is not ready.
+ * @retval -ENOTSUP if CONFIG_CRC is not enabled.
+ * @retval -EOVERFLOW if offset + size would result in an integer overflow.
+ *
+ * @see crc32_ieee_update
+ */
+int flash_region_crc(const struct device *dev, off_t offset, off_t size, uint32_t *crc,
```

Comment:
I am not convinced regarding passing the user buffer.
What would be the use case? Re-use of some buffer user already has?

Reply:
removed